plugin/fzf.vim

```diff
@@ -44,20 +44,31 @@ if s:is_win
       let &shellslash = shellslash
     endtry
   endfunction
-
-  function! s:fzf_shellescape(path)
-    return substitute(s:fzf_call('shellescape', a:path), '[^\\]\zs\\"$', '\\\\"', '')
-  endfunction
 else
   function! s:fzf_call(fn, ...)
     return call(a:fn, a:000)
   endfunction
-
-  function! s:fzf_shellescape(path)
-    return shellescape(a:path)
-  endfunction
 endif
 
+function! s:shellesc_sh(arg)
+  return '"'.substitute(a:arg, '"', '\\"', 'g').'"'
```

Comment:
I can't test in a linux box atm but I had no issues with this change in cygwin (using ConEmu, not mintty). It didn't collide with `s:escape`. Tested with this branch, https://github.com/janlazo/fzf/tree/vim/shellesc_cygwin_sh

If there's no issues on your end (including fzf.vim), I'll commit your suggestion (not the cygwin changes) and update the test cases.